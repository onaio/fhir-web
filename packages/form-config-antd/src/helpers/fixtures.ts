/* eslint-disable @typescript-eslint/camelcase */
export const manifestRelease4 = {
  identifier: '1.0.30',
  json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
  appId: 'org.smartregister.foo',
  appVersion: '1.2.18',
  createdAt: '2020-06-09T14:40:04.189+02:00',
  updatedAt: '2020-06-09T14:40:04.189+02:00',
};

export const fixManifestReleases = [
  {
    identifier: '1.0.11',
    json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:00:38.413+02:00',
    updatedAt: '2020-06-08T15:00:38.413+02:00',
  },
  {
    identifier: '1.0.12',
    json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:03:00.950+02:00',
    updatedAt: '2020-06-08T15:03:00.950+02:00',
  },
  {
    identifier: '1.0.16',
    json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.16',
    createdAt: '2020-06-09T14:40:04.189+02:00',
    updatedAt: '2020-06-09T14:40:04.189+02:00',
  },
];

export const manifestFile1 = {
  createdAt: 'Jun 19, 2020, 12:31:37 PM',
  form_relation: '',
  id: '52',
  identifier: 'test-form-1.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test form',
  module: '',
  version: '1.0.26',
};

export const manifestFile2 = {
  createdAt: 'Jun 19, 2020, 4:23:22 PM',
  form_relation: '',
  id: '53',
  identifier: 'reveal-test-file.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test publish',
  module: '',
  version: '1.0.27',
};

export const manifestFile3 = {
  createdAt: 'Jun 19, 2020, 6:23:22 PM',
  form_relation: '',
  id: '54',
  identifier: 'foo.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'foo',
  module: 'baz',
  version: '1.0.28',
};

export const draftFile1 = {
  ...manifestFile1,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile2 = {
  ...manifestFile2,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile3 = {
  ...manifestFile3,
  isDraft: true,
  isJsonValidator: false,
};

export const fixManifestFiles = [manifestFile1, manifestFile2];

export const FixManifestDraftFiles = [
  {
    ...fixManifestFiles[0],
    isDraft: false,
  },
  {
    ...fixManifestFiles[1],
    isDraft: false,
  },
];

export const downloadFile = {
  clientForm: {
    json: JSON.stringify(fixManifestFiles[0]),
  },
};
