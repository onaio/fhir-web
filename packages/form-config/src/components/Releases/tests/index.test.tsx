import React from 'react';
import { mount, shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import { ManifestReleases, ConnectedManifestReleases } from '../index';
import { getFetchOptions, OpenSRPService } from '@opensrp/server-service';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import reducerRegistry, { store } from '@onaio/redux-reducer-registry';
import flushPromises from 'flush-promises';
import releasesReducer, {
  fetchManifestReleases,
  releasesReducerName,
} from '../../../ducks/manifestReleases';
import { fixManifestReleases } from '../../../ducks/tests/fixtures';
import toJson from 'enzyme-to-json';
import _ from 'lodash';

/** register the reducers */
reducerRegistry.register(releasesReducerName, releasesReducer);

const history = createBrowserHistory();

const baseURL = 'https://test-example.com/rest';
const endpoint = 'manifest';
const props = {
  baseURL,
  currentUrl: '/releases',
  endpoint,
  formUploadUrl: '/upload',
  getPayload: getFetchOptions,
  LoadingComponent: <div>Loading</div>,
  uploadTypeUrl: 'file',
};

const actualDebounce = _.debounce;
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customDebounce = (callback: any) => callback;
_.debounce = customDebounce;

describe('components/ManifestReleases', () => {
  afterAll(() => {
    _.debounce = actualDebounce;
  });

  it('renders without crashing', () => {
    shallow(<ManifestReleases {...props} />);
  });

  it('renders without crashing when connected to store', async () => {
    store.dispatch(fetchManifestReleases(fixManifestReleases));
    const mockList = jest.fn();
    OpenSRPService.prototype.list = mockList;
    mockList.mockReturnValueOnce(Promise.resolve(fixManifestReleases));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedManifestReleases {...props} />
        </Router>
      </Provider>
    );

    await flushPromises();
    wrapper.update();
    expect(wrapper.find('DrillDownTable').props()).toMatchSnapshot();
    expect(wrapper.find('SearchBar')).toHaveLength(1);
    expect(wrapper.find('Row Col').at(1).text()).toEqual('Upload New File');

    expect(wrapper.find('.tbody .tr')).toHaveLength(fixManifestReleases.length);

    const viewFiledCell = wrapper.find('.tbody .tr').at(0).find('.td').at(4).find('a');
    expect(toJson(viewFiledCell)).toMatchSnapshot();
    expect(viewFiledCell.text()).toEqual('View Files');
  });

  it('test search', async () => {
    const mockList = jest.fn();
    OpenSRPService.prototype.list = mockList;
    mockList.mockReturnValueOnce(Promise.resolve(fixManifestReleases));

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedManifestReleases {...props} />
        </Router>
      </Provider>
    );
    await flushPromises();
    wrapper.update();

    // search
    const search = wrapper.find('SearchBar input');
    search.simulate('input', { target: { value: '1.0.12' } });
    wrapper.update();
    expect(wrapper.find('.tbody .tr')).toHaveLength(1);
  });
});
