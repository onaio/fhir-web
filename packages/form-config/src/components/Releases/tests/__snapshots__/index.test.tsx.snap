// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ManifestReleases renders without crashing when connected to store 1`] = `
Object {
  "CellComponent": [Function],
  "columns": Array [
    Object {
      "Header": "Identifier",
      "accessor": [Function],
    },
    Object {
      "Header": "App Id",
      "accessor": "appId",
    },
    Object {
      "Header": "App Version",
      "accessor": [Function],
    },
    Object {
      "Cell": [Function],
      "Header": "Updated at",
      "accessor": "updatedAt",
    },
    Object {
      "Header": " ",
      "accessor": [Function],
      "disableSortBy": true,
    },
  ],
  "data": Array [
    Object {
      "appId": "org.smartregister.reveal",
      "appVersion": "1.2.16",
      "createdAt": "2020-06-09T14:40:04.189+02:00",
      "identifier": "1.0.16",
      "json": "{\\"forms_version\\":\\"1.0.16\\",\\"identifiers\\":[\\"blood_screening.json\\"}",
      "updatedAt": "2020-06-09T14:40:04.189+02:00",
    },
    Object {
      "appId": "org.smartregister.reveal",
      "appVersion": "1.2.14",
      "createdAt": "2020-06-08T15:03:00.950+02:00",
      "identifier": "1.0.12",
      "json": "{\\"forms_version\\":\\"1.0.12\\",\\"identifiers\\":[\\"blood_screening.json\\"}",
      "updatedAt": "2020-06-08T15:03:00.950+02:00",
    },
    Object {
      "appId": "org.smartregister.reveal",
      "appVersion": "1.2.14",
      "createdAt": "2020-06-08T15:00:38.413+02:00",
      "identifier": "1.0.11",
      "json": "{\\"forms_version\\":\\"1.0.11\\",\\"identifiers\\":[\\"blood_screening.json\\"}",
      "updatedAt": "2020-06-08T15:00:38.413+02:00",
    },
  ],
  "hasChildren": [Function],
  "identifierField": "id",
  "linkerField": "id",
  "loading": false,
  "loadingComponent": [Function],
  "paginate": false,
  "parentIdentifierField": "parent_id",
  "renderNullDataComponent": [Function],
  "resize": true,
  "rootParentId": null,
  "useDrillDown": false,
}
`;

exports[`components/ManifestReleases renders without crashing when connected to store 2`] = `
<a
  href="/releases/1.0.16"
  onClick={[Function]}
>
  View Files
</a>
`;
